ui/src/components/admin/leadershipTeam/AdminLeadershipTeamContainer.js

```diff
@@ -1,125 +1,368 @@
+      if (data) setTeams(data);
     } catch (err) {
       console.log(err);
     }
   }, []);
 
   useEffect(() => {
     getData();
-    checkIfUpdated(false);
-    setInterval(() => {
-      checkIfUpdated();
-    }, pollFreqInSecs * 1000);
-  }, [checkIfUpdated]);
-
-  const defaultColDef = {
-    editable: true,
-    sortable: true,
-    resizable: true,
-    filter: true,
-    floatingFilter: true,
-  };
+  }, [getData]);
 
-  // handle export of csv
-  const exportParams = () => {
-    let fileName = 'leadership_team.csv';
-    if (lastUpdatedTime.current) {
-      const asOfDate = lastUpdatedTime.current.toFormat('yyyyMMdd_HHmmss');
-      fileName = `${asOfDate}_leadership_team.csv`;
+  useEffect(() => {
+    if (selected) {
+      setId(selected.id);
+      setSeasonFrom(selected.seasonFrom);
+      setSeasonTo(selected.seasonTo);
+      setCampus(selected.campus);
+      setLifestage(selected.lifestage);
+      setLifeGroup(selected.lifeGroup);
+      setLeaders(selected.leaders);
+      setDeleted(selected.isDeleted);
     }
+  }, [selected, setLeaders]);
+
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const firstField = useRef();
 
-    return {
-      skipColumnGroupHeaders: true,
-      allColumns: true,
-      fileName: fileName,
-    };
+  const createHandler = async () => {
+    try {
+      const leadersArr = leaders && leaders.length > 0 && leaders.split(',');
+      const cleanLeadersArr = leadersArr.map((x) => x.trim());
+
+      const res = await axios.post('/api/leadership-team/create', {
+        seasonFrom,
+        seasonTo,
+        campus,
+        lifestage,
+        lifeGroup,
+        leaders: cleanLeadersArr,
+        isDeleted: deleted,
+      });
+
+      if (res.status === 200) return true;
+    } catch (e) {
+      console.log(e.response);
+      toast({
+        description: e.response.data,
+        status: 'error',
+        duration: 5000,
+      });
+      return false;
+    }
   };
 
-  const exportHandler = () => {
-    if (api) {
-      api.exportDataAsCsv(exportParams());
+  const updateHandler = async () => {
+    try {
+      const leadersArr = leaders && leaders.length > 0 && leaders.split(',');
+      const cleanLeadersArr = leadersArr.map((x) => x.trim());
+
+      const res = await axios.put('/api/leadership-team/update', {
+        id,
+        seasonFrom,
+        seasonTo,
+        campus,
+        lifestage,
+        lifeGroup,
+        leaders: cleanLeadersArr,
+        isDeleted: deleted,
+      });
+
+      if (res.status === 200) return true;
+    } catch (e) {
+      console.log(e.response);
+      toast({
+        description: e.response.data,
+        status: 'error',
+        duration: 5000,
+      });
+      return false;
     }
   };
 
-  useEffect(() => {
-    if (api) {
-      if (teams && teams.length) {
-        api.hideOverlay();
-      } else {
-        api.showLoadingOverlay();
+  const duplicateHandler = async (e) => {
```

Comment:
the duplicate functionality sets the last updated by to be by the original modifier. e.g. if original one says "Tamara" after duplicate it will say "Tamara" too. Should this be changed to the user that initiates the duplicate operation?